src/App.jsx

```diff
@@ -14,45 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ABC = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+
+const getPreparedGoods = (goods, sortGoods, isReverse) => {
+  const preparedGoods = [...goods];
+
+  if (sortGoods) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortGoods) {
+        case SORT_ABC:
+          return good1.localeCompare(good2);
+
+        case SORT_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortGoods, setSortGoods] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortGoods,
+    isReverse,
+  );
+
+  const changeSortGoods = (typeSort) => {
+    setSortGoods(typeSort);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            ' is-light': sortGoods !== SORT_ABC,
+          })}
+          onClick={() => changeSortGoods(SORT_ABC)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            ' is-light': sortGoods !== SORT_LENGTH,
+          })}
+          onClick={() => changeSortGoods(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            ' is-light': !isReverse,
+          })}
+          onClick={() => setIsReverse(!isReverse)}
+        >
+          Reverse
+        </button>
+
+        {(sortGoods || isReverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              changeSortGoods('');
+              setIsReverse(false);
+            }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
            onClick={onReset}
```